ios/PurchasesHybridCommon/PurchasesHybridCommonTests/PurchaserInfoHybridAdditionsTests.swift

```diff
@@ -8,95 +8,97 @@
+                                        "id": "transactionid",
+                                        "is_sandbox": true,
+                                        "original_purchase_date": "1990-08-30T02:40:36Z",
+                                        "purchase_date": transactionDateString,
+                                        "store": "play_store"
+                                    ]
+                                ]
+                            ]
+                        ]])
+
+                    let dateformatter = ISO8601DateFormatter()
+                    let transactionDate = dateformatter.date(from: transactionDateString)!
+
+                    let dictionary = mockCustomerInfo.dictionary
                     let nonSubscriptionTransactions = dictionary["nonSubscriptionTransactions"] as? Array<Any>
                     expect(nonSubscriptionTransactions?.count) == 1
-                    
+
                     let transactionDictionary = nonSubscriptionTransactions?[0] as? Dictionary<String, Any>
-                    expect(transactionDictionary?["revenueCatId"] as? String) == transaction.revenueCatId
-                    expect(transactionDictionary?["productId"] as? String) == transaction.productId
-                    expect(transactionDictionary?["purchaseDateMillis"] as? Double) == (transactionDate as NSDate).rc_millisecondsSince1970AsDouble()
-                    
-                    let dateformatter = ISO8601DateFormatter()
+                    expect(transactionDictionary?["revenueCatId"] as? String) == "transactionid"
+                    expect(transactionDictionary?["productId"] as? String) == "productid"
+                    expect(transactionDictionary?["purchaseDateMillis"] as? Double) == transactionDate.rc_millisecondsSince1970AsDouble()
+
+
                     expect(transactionDictionary?["purchaseDate"] as? String) == dateformatter.string(from: transactionDate as Date)
                 }
                 it ("is empty when there are no non subscription transactions") {
-                    let purchaserInfo = PartialMockPurchaserInfo()
-                
-                    let dictionary = purchaserInfo.dictionary
+                    let mockCustomerInfo = try! CustomerInfo(data: [
+                        "request_date": "2019-08-16T10:30:42Z",
+                        "subscriber": [
+                            "first_seen": "2019-07-17T00:05:54Z",
+                            "original_app_user_id": "",
+                            "subscriptions": [:],
+                            "other_purchases": [:]
+                        ]])
+
+                    let dictionary = mockCustomerInfo.dictionary
                     let nonSubscriptionTransactions = dictionary["nonSubscriptionTransactions"] as? Array<Any>
                     expect(nonSubscriptionTransactions?.count) == 0
                 }
             }
         }
     }
 }
-
-class PartialMockPurchaserInfo: Purchases.PurchaserInfo {
```

Comment:
this approach no longer worked with v4 because you can't override `let` properties in a subclass, so the approach (that's taken from https://github.com/RevenueCat/purchases-hybrid-common/pull/113) just creates purchaserInfo directly

Reply:
Wouldn't be a problem if we could make `CustomerInfo` a `struct` with `var`s 😢

Reply:
the sadness. 
Maybe for the next major we can start thinking about exposing a different API for swift and ObjC and get more flexibility for swift-only stuff? It might get painful to maintain, though